### PR TITLE
Move common Makefile targets to Makefile.common

### PR DESCRIPTION
**Pull request by rafaelvanoni**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,132 +1,36 @@
-# Shortcut targets
-default: build
-
-## Build binary for current platform
-all: build
-
-## Run the tests for the current platform/architecture
-test: ut
+PACKAGE_NAME?=github.com/projectcalico/node
```
Comment (realgaurav) [inline Makefile]: Why is this `node`, for my make benefit?

File: Makefile [hunk 2]
```diff
@@ -1,132 +1,36 @@
...
-		-v $(CURDIR)/.go-pkg-cache:/go-cache:rw \
-		-w /go/src/$(PACKAGE_NAME)
-
 .PHONY: clean
 ## Clean enough that a new release build will be clean
 clean:
 	find . -name '*.created-$(ARCH)' -exec rm -f {} +
-	rm -rf report/￼
+	rm -rf report/
+	rm -rf bin/flexvol-$(ARCH)
```
Comment (realgaurav) [inline Makefile]: Is this repo specific improvement? I can't seem to relate it to Makefile changes intended  here.

File: Makefile (hunk 1 above)
Comment (rafaelvanoni) [reply Makefile]: That's a bug :-) Fixed

File: Makefile (hunk 2 above)
Comment (rafaelvanoni) [reply Makefile]: Yes, that object was missing from the `clean` target.

**Pull request merged**